[germainelee02] iP

germainelee02 opened the pull request:
# Duke
## CS2103T iP
> This is a random quote about software engineering. ([source](https://medium.com/geekculture/famous-quotes-about-software-engineer-everyone-should-know-9b946cd6ec66#:~:text=%E2%80%9CSoftware%20and%20cathedrals%20are%20much,language.%E2%80%9D%20by%20Larry%20Wall.))
- this
- is
- a
- ~~bullet~~
- list
- [ ] this
- [ ] is
- [ ] *a*
- [ ] task 
- [X] list
1. this
2. is
3. a
4. **numbered**
5. list
6. EMOJI 💯 

`System.out.println("inline code");`



Check out this Java `main` method
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

germainelee02 committed:
Level-1

germainelee02 committed:
Level-2

germainelee02 committed:
Level-3

germainelee02 committed:
Level-3

germainelee02 committed:
added subclasses for task

germainelee02 committed:
added tests

germainelee02 committed:
exceptions

germainelee02 committed:
Level-6: added delete

germainelee02 committed:
added list update string

germainelee02 committed:
Level-6 fix bugs

germainelee02 committed:
toString()

germainelee02 committed:
enums

germainelee02 committed:
added javadocs

germainelee02 committed:
Level-7

germainelee02 committed:
Level-7

germainelee02 committed:
add level-8 tasks

germainelee02 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

germainelee02 committed:
change date time format

germainelee02 committed:
add more oop

germainelee02 committed:
encapsulate parser

germainelee02 committed:
add packages

germainelee02 committed:
Merge branch 'add-gradle-support' of https://github.com/germainelee02/ip

germainelee02 committed:
add gradle

germainelee02 committed:
add junit tests

germainelee02 committed:
add javadoc

germainelee02 committed:
add javadocs

germainelee02 committed:
fix coding standard

germainelee02 committed:
add javadocs

germainelee02 committed:
fix coding standard

germainelee02 committed:
fix bug

germainelee02 committed:
level-9

germainelee02 committed:
changed formatter

germainelee02 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/taskers/Deadline.java
#	src/main/java/duke/taskers/Event.java
#	src/main/java/duke/taskers/Task.java
#	src/main/java/duke/taskers/Todo.java
#	src/main/java/duke/util/Storage.java
#	src/main/java/duke/util/Ui.java

germainelee02 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/data/duke
#	src/main/java/duke/Duke.java
#	src/main/java/duke/taskers/Deadline.java
#	src/main/java/duke/taskers/Event.java
#	src/main/java/duke/util/Parser.java
#	src/main/java/duke/util/TaskList.java

germainelee02 committed:
fix javadocs

germainelee02 committed:
fix packaging

germainelee02 committed:
Delete src/main/data directory

germainelee02 committed:
Delete text-ui-test/src/main/data directory

germainelee02 committed:
fixes pacakge and text ui testing

germainelee02 committed:
add config

germainelee02 committed:
add more oop

germainelee02 committed:
fix breaking text-ui-test

germainelee02 committed:
error handling

germainelee02 committed:
Update README.md

germainelee02 committed:
Delete text-ui-test/src/main/data directory

germainelee02 committed:
Delete src/main/data directory

germainelee02 committed:
add gui

germainelee02 committed:
Merge branch 'master' of https://github.com/germainelee02/ip

# Conflicts:
#	src/main/data/duke

jiasheng59 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import duke.commands.Command;
+import duke.exceptions.DukeException;
+import duke.utils.Parser;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+import duke.utils.Ui;
```
Perhaps can take some time to reorder the import statements

jiasheng59 commented on src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,136 @@
+     */
+    public static String taskStringFormatter(Task t) {
+        String s = "";
+        if (t instanceof Todo) {
+            Todo todo = (Todo) t;
+            s = todo.formatStringForFile();
+        } else if (t instanceof Deadline) {
+            Deadline deadline = (Deadline) t;
+            s = deadline.formatStringForFile();
+        } else {
+            Event event = (Event) t;
+            s = event.formatStringForFile();
+        }
```
Is there any more elegant way of writing this part?

jiasheng59 commented on src/main/java/duke/utils/Ui.java:
```diff
@@ -0,0 +1,167 @@
+    /**
+     * Duke greets when the program first starts.
+     */
+    public String greet() {
+        return "     Hello! I am Duke!\n     What can I do for you today?";
+    }
```
Perhaps can abstract out the indentation spaces into a variable

jiasheng59 commented on src/main/java/duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,49 @@
+     * Checks if command exits or not.
+     * @return Supposed to return false.
+     */
+    @Override
+    public boolean isExit() {
+        return false;
+    }
```
Good to see you are following the naming convention

jiasheng59 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,103 @@
+    public static LocalDateTime createLocalDateTime(String dateTime) {
+        LocalDateTime date;
+        try {
+            String stringWithNoTrailingWhitespaces = dateTime.trim();
```
I like the self-explanatory name for this variable

jiasheng59 reviewed (commented):
Overall, there's no big coding standard issues. LGTM

germainelee02 committed:
add gui

germainelee02 committed:
Add assertions.

May not have an easy way for the coder to check that program behaviour is to be expected. Bugs may not be as easy to find.

As we continue to grow this Duke project, we need an easier way for programmers to detect bugs or areas where it is not performing as desired. This is especially when the code base is continually being modified.

It is done this way because it is simple. Assertions also stops the program flow when a bug is detected so as to not cause further complications.

Refer to CS2103T Textbook on assertions to see why assertions are important

germainelee02 committed:
Fix code quality

Code quality is already quite good, but cleaned up some things for better readability

No need to instantiate variables as null as they are already null by default during declaration.

Changing it up will improve readability

Check out CS2103T textbook on code quality to understand more

germainelee02 committed:
Merge pull request #2 from germainelee02/Branch-A-Assertions

Add assertions.

germainelee02 committed:
Merge pull request #3 from germainelee02/Branch-A-CodeQuality

Fix code quality

germainelee02 committed:
Add extra prioritising functionality.

If the list gets long, users will not be able to focus on what is really important to them, hence they need to prioritise some tasks.

Prioritising is just an additional command that is easy to use and pick up.

Read up on prioritising tasks at www.google.com.

germainelee02 committed:
deleted duke file

germainelee02 committed:
add pages readme and screenshot of app

germainelee02 committed:
improved docs

germainelee02 committed:
fix finding bug. removed some random print statements
